### PR TITLE
tutorials: add a tutorial on submitting jobs to Flux

### PR DESCRIPTION
**Pull request by cmoussa1**
This is a small [WIP] PR built on top of #192 that adds a tutorial on how to submit jobs to Flux. It leverages the steps outlined in the [job-submit-cli](https://github.com/flux-framework/flux-workflow-examples/tree/master/job-submit-cli) workflow example. I ran the commands outlined in an updated Docker container to make sure they still worked (I could always use another set of eyes to double-check me, though 😉).

Future expansion of this specific tutorial could include a chapter on how to submit jobs to Flux using it's job submission API. But for the purposes of getting some short tutorials out there, I've just included the command-line portion.

### CODE REVIEW

**Comment by chu11** [comment]
immediate thought is if we should cover lots of variants and common options:

- `--task-per-node` and `--tasks-per-core`?
- `--wait` option?
- `flux mini run` - run until completion, stdout/stderr live
- batch / alloc
- launch to get a shell (alloc)
- `flux job attach`??
- `flux job cancel`?
- `flux job kill`?
- job dependencies (although this might be under its own tutorial)?

so this might become more more "job submission and management basics"??

**Comment by garlick** [comment]
Maybe this could be one of those documents like you proposed @chu11 where it starts simple and adds complexity?  If the topic is "job submission" Maybe it should try to stay focused on getting jobs into the system?

**Comment by chu11** [comment]
> Maybe this could be one of those documents like you proposed @chu11 where it starts simple and adds complexity? If the topic is "job submission" Maybe it should try to stay focused on getting jobs into the system?

Good point, perhaps this specific doc should be simpler then, no need to explain getting `R` and stuff at the bottom.

**Comment by grondo** [comment]
This one is under "command tutorials". So maybe `flux mini` tutorial or `flux mini submit` tutorial is the correct title. Though it is admittedly very difficult to keep to that specific topic in a tutorial (kind of feel like that is what command man pages should be?)

File: tutorials/commands/index.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
+.. _command-tutorials:
+
+Command Tutorials
+=================
+
+Welcome to the Command Tutorials! These tutorials should help you to map specific Flux commands
+with your use case, and then see detailed usage. 
+
+ - ``flux proxy`` (:ref:`ssh-across-clusters`): "Send commands to a flux instance across clusters using ssh"
+ - ``job-submit`` (:ref:`job-submit`): "Submit a job in a Flux instance"
```
Comment (vsoch) [inline tutorials/commands/index.rst:10]: Should this be `flux mini submit` ? I think here we want to start to help to make associations between the actual command to be run and the use case.

File: tutorials/commands/job-submit.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
+.. _job-submit:
```
Comment (vsoch) [inline tutorials/commands/job-submit.rst:1]:
```suggestion
.. _flux-mini-submit:
```

File: tutorials/commands/job-submit.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+options:
+
+.. code-block:: sh
+
+    $ flux mini submit --nodes=2 --ntasks=4 --cores-per-task=2 ./my_compute_script.lua 120
+    ƒM5k8m7m
+    $ flux mini submit --nodes=1 --ntasks=1 --cores-per-task=2 ./my_other_script.lua 120
+    ƒSUEFPDH
+
+A jobID is returned for every job submitted. You can view the status of your
```
Comment (vsoch) [inline tutorials/commands/job-submit.rst:24]:
```suggestion
A jobID (e.g., ``ƒSUEFPDH``) is returned for every job submitted. You can view the status of your
```
And do we have a term for this in the new terms guide? If yes - let's link!

File: tutorials/commands/job-submit.rst [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+       JOBID  USER     NAME       ST NTASKS NNODES     TIME INFO
+    ƒSUEFPDH  fluxuser my_other_s  R      1      1   1.842s 
+    ƒM5k8m7m  fluxuser my_compute  R      4      2   3.255s
+
+You can also get information about your job by passing in the jobID and a key.
+For example, to get resource information about your job:
+
+.. code-block:: sh
+
+  $ flux job info ƒM5k8m7m R
```
Comment (vsoch) [inline tutorials/commands/job-submit.rst:39]: Is there a way to get this in a table (non json) - this was a question I had the other day and I couldn't figure out from the command line.

**Review by vsoch — COMMENTED** [review]
some comments! And I guess this depends on merging the other ssh PR?

File: tutorials/commands/index.rst (hunk 1 above)
Comment (grondo) [reply tutorials/commands/index.rst:10]: If I understand what you mean, I think the other PR needs to be merged first. This PR just contains the other PR's commits which is why the file is new.

File: tutorials/commands/job-submit.rst (hunk 4 above)
Comment (grondo) [reply tutorials/commands/job-submit.rst:39]: I think actually we should not advertise `flux job info` in a high-level tutorial. It is more of a "plumbing" command. Instead we should focus on `flux jobs`, the main interface users will use to get information about their jobs.

File: tutorials/commands/job-submit.rst [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  {"version":1,"execution":{"R_lite":[{"rank":"0-1","children":{"core":"0-3"}}]}}
+
+There are a number of keys you can pass to get various information about your job:
```
[the remark this answers is not in the document]
Comment (grondo) [reply tutorials/commands/job-submit.rst:42]: This is part of the reason why it is probably best to leave `flux job info` out of user tutorials. An advanced tutorial could perhaps explain this command

[the remark this answers is not in the document]
Comment (garlick) [reply tutorials/commands/job-submit.rst:42]: `flux job info` is a kind of plumbing command, not very user friendly, and probably shouldn't be here IMHO

File: tutorials/commands/index.rst (hunk 1 above)
Comment (vsoch) [reply tutorials/commands/index.rst:10]: Yep, that is what I was trying to say, poorly.

File: tutorials/commands/job-submit.rst [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+Submit Jobs Using Flux's CLI
+----------------------------
+
+From within a Flux instance, you can submit your job on the command line with
+optional arguments, resource options, per task options, and per resource
+options:
+
+.. code-block:: sh
+
+    $ flux mini submit --nodes=2 --ntasks=4 --cores-per-task=2 ./my_compute_script.lua 120
```
Comment (chu11) [inline tutorials/commands/job-submit.rst:19]: was thinking if there are any other important options to mention, the only one I could think of is `--queue`.

IMO, that's more important than `--cores-per-task`.  You could probably just mention that there are many advanced ways to request resources besides `--nodes` and `--ntasks`, and to see the manpage for details.

Comment (vsoch) [reply tutorials/commands/job-submit.rst:19]: If you are open to it, I like having (toward the end) a big block of just examples with description, that sort of show all the options that a flux command can provide. E.g., I think I linked this before, but this example comes to mind! https://rse-ops.github.io/knowledge/docs/schedulers/slurm.html#command-quick-reference. Sometimes people's eyes will glaze over the text and they just want to find the right thing to copy paste :) (guilty!)

Comment (chu11) [reply tutorials/commands/job-submit.rst:19]: oh that's a good idea

File: tutorials/commands/job-submit.rst (hunk 4 above)
Comment (cmoussa1) [reply tutorials/commands/job-submit.rst:39]: Oops! Sorry for including it in this tutorial. I'll go ahead and just remove this section then and only include `flux jobs`.

**Comment by cmoussa1** [comment]
Thanks for all the great feedback and suggestions everybody! I've just force pushed some changes to the tutorial based on the suggestions above. To summarize, I've made the following changes:

1) I've changed the name of the document and its title to `flux-mini-submit`
2) I've included a short description of the two job submission examples at the top of the tutorial and provided links to both the definition of a Flux job ID (to our new glossary) and the `flux mini` man page.
3) I've removed references to `flux job info` and just included `flux jobs`.
4) I've attached a couple more examples to submitting jobs using `flux mini` at the bottom of the tutorial that use some different options. Specifically, I've added an example submitting a job to a specific queue and an example submitting a job with `--dry-run` to get the jobspec for the job (and linking to the definition of jobspec in our glossary).

Thanks again for the feedback. I think I might as well take this out of [WIP], but note that we should probably land #192 before this one, since this PR is built on top of #192.

File: tutorials/commands/flux-mini-submit.rst [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+.. code-block:: sh
+
+    $ flux mini submit --nodes=2 --queue=foo --name=my_special_job ./my_job.lua
+
+This submits a job to the `foo` queue across two nodes, and sets a custom name
+to the job.
+
+.. code-block:: sh
+
+    $ flux mini submit --dry-run ./my_cool_job.lua
```
Comment (chu11) [inline tutorials/commands/flux-mini-submit.rst:59]: i like this.  I will suggest we have one example with the `--output` option, as I imagine many would want that.  It'd perhaps be wise to illustrate use of `{{id}}` in the output option too.

Comment (cmoussa1) [reply tutorials/commands/flux-mini-submit.rst:59]: Oh that's a good idea, thanks. I'll add an example that includes this!

Comment (cmoussa1) [reply tutorials/commands/flux-mini-submit.rst:59]: Just force-pushed a commit that adds an example including `--output` and `{{id}}` when submitting a job.

File: tutorials/commands/flux-mini-submit.rst [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+Submit Jobs Using Flux's CLI
+----------------------------
+
+From within a Flux instance, you can submit your job on the command line with
+optional arguments, resource options, per task options, and per resource
+options:
+
+.. code-block:: sh
+
+    $ flux mini submit --nodes=2 --ntasks=4 --cores-per-task=2 ./my_compute_script.lua 120
```
Comment (chu11) [inline tutorials/commands/flux-mini-submit.rst:19]: minor nit, should we do `.py` everywhere instead of `.lua`?  since python is more popular.

Comment (cmoussa1) [reply tutorials/commands/flux-mini-submit.rst]: Oh, sure! Good call. Just force-pushed a fix to use `.py` instead of `.lua`

**Comment by vsoch** [comment]
@cmoussa1 the ssh tutorial is merged! You should be able to rebase locally and then we can finish up review here and get the tutorial in.

**Review by chu11 — APPROVED** [review]
LGTM!

**Comment by cmoussa1** [comment]
Thanks @vsoch and @chu11! 😃 I've just rebased off current `master` and will set MWP on this

**Pull request merged**